### PR TITLE
ovl_Dm_Char06 OK

### PR DESCRIPTION
**Pull request by ordlucas**

### CODE REVIEW

File: src/overlays/actors/ovl_Dm_Char06/z_dm_char06.c [hunk 1]
```diff
@@ -32,18 +30,60 @@ const ActorInit Dm_Char06_InitVars = {
...
+    this->alpha = 255;
+    DmChar06_SetupAction(this, func_80AAE6F0);
+}
+
+void DmChar06_Destroy(Actor* thisx, PlayState* play) {
+}
+
+void func_80AAE6F0(DmChar06* this, PlayState* play) {
+    if (Cutscene_CheckActorAction(play, 463)) {
+        s32 actionIndex = Cutscene_GetActorActionIndex(play, 463);
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_Dm_Char06/z_dm_char06.c:51]: We've been using hex for these so far.

File: src/overlays/actors/ovl_Dm_Char06/z_dm_char06.c [hunk 2]
```diff
@@ -32,18 +30,60 @@ const ActorInit Dm_Char06_InitVars = {
...
+
+        if (play->csCtx.frames >= play->csCtx.actorActions[actionIndex]->startFrame &&
+            play->csCtx.actorActions[actionIndex]->endFrame >= play->csCtx.frames) {
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_Dm_Char06/z_dm_char06.c:54]:
```suggestion
        if ((play->csCtx.frames >= play->csCtx.actorActions[actionIndex]->startFrame) &&
            (play->csCtx.actorActions[actionIndex]->endFrame >= play->csCtx.frames)) {
```

File: src/overlays/actors/ovl_Dm_Char06/z_dm_char06.c [hunk 3]
```diff
@@ -32,18 +30,60 @@ const ActorInit Dm_Char06_InitVars = {
...
+            if (play->csCtx.actorActions[actionIndex]->action == 1) {
+                this->alpha = 255;
+            } else if (play->csCtx.actorActions[actionIndex]->action == 2) {
+                f32 floatTemp = 1.0f - Environment_LerpWeight(play->csCtx.actorActions[actionIndex]->endFrame,
+                                                              play->csCtx.actorActions[actionIndex]->startFrame,
+                                                              play->csCtx.frames);
+                this->alpha = (u8)(255.0f * floatTemp);
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_Dm_Char06/z_dm_char06.c:61]: Cast required?

File: src/overlays/actors/ovl_Dm_Char06/z_dm_char06.c [hunk 4]
```diff
@@ -32,18 +30,60 @@ const ActorInit Dm_Char06_InitVars = {
...
+    DmChar06* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Char06/DmChar06_SetupAction.s")
+    this->actionFunc(this, play);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Char06/DmChar06_Init.s")
+void DmChar06_Draw(Actor* thisx, PlayState* play) {
+    DmChar06* this = THIS;
+    s32 pad;
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_Dm_Char06/z_dm_char06.c:75]: Put the pad above if possible.

File: src/overlays/actors/ovl_Dm_Char06/z_dm_char06.h [hunk 5]
```diff
@@ -2,16 +2,18 @@
...
+#include "assets/objects/object_yukiyama/object_yukiyama.h"
 
 struct DmChar06;
 
 typedef void (*DmChar06ActionFunc)(struct DmChar06*, PlayState*);
 
 typedef struct DmChar06 {
     /* 0x000 */ Actor actor;
-    /* 0x144 */ char unk_144[0x14C];
+    /* 0x144 */ UNK_TYPE1 pad144[0x14C];
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_Dm_Char06/z_dm_char06.h:13]: Goodness that's a lot of padding.

File: src/overlays/actors/ovl_Dm_Char06/z_dm_char06.c [hunk 6]
```diff
@@ -32,18 +30,60 @@ const ActorInit Dm_Char06_InitVars = {
...
+    Actor_SetScale(&this->actor, 1.0f);
+    this->alpha = 255;
+    DmChar06_SetupAction(this, func_80AAE6F0);
+}
+
+void DmChar06_Destroy(Actor* thisx, PlayState* play) {
+}
+
+void func_80AAE6F0(DmChar06* this, PlayState* play) {
+    if (Cutscene_CheckActorAction(play, 463)) {
```
Comment (hensldm) [inline src/overlays/actors/ovl_Dm_Char06/z_dm_char06.c:50]:
```suggestion
    if (Cutscene_CheckActorAction(play, 0x1CF)) {
```

File: src/overlays/actors/ovl_Dm_Char06/z_dm_char06.c [hunk 7]
```diff
@@ -32,18 +30,60 @@ const ActorInit Dm_Char06_InitVars = {
...
+        s32 actionIndex = Cutscene_GetActorActionIndex(play, 0x1CF);
+
+        if ((play->csCtx.frames >= play->csCtx.actorActions[actionIndex]->startFrame) &&
+            (play->csCtx.actorActions[actionIndex]->endFrame >= play->csCtx.frames)) {
+            if (play->csCtx.actorActions[actionIndex]->action == 1) {
+                this->alpha = 255;
+            } else if (play->csCtx.actorActions[actionIndex]->action == 2) {
+                f32 floatTemp = 1.0f - Environment_LerpWeight(play->csCtx.actorActions[actionIndex]->endFrame,
```
Comment (hensldm) [inline src/overlays/actors/ovl_Dm_Char06/z_dm_char06.c:58]:
```suggestion
                f32 lerp= 1.0f - Environment_LerpWeight(play->csCtx.actorActions[actionIndex]->endFrame,
```
I think lerp would be a better variable name

**Review by hensldm — APPROVED** [review]

**Review by EllipticEllipsis — APPROVED** [review]

**Review by jpburnett — APPROVED** [review]

**Review by Kenix3 — APPROVED** [review]

**Pull request merged**